templates/token-staking-dapp-template/frontend/providers/accountData.tsx

```diff
@@ -0,0 +1,122 @@
+import { useToast } from "@/components/ui/use-toast";
+import { useGetTokenData } from "@/hooks/useGetTokenData";
+import { convertAmountFromOnChainToHumanReadable } from "@/utils/helpers";
+import { getAccountTokenBalance } from "@/view-functions/getAccountTokenAmount";
+import { getClaimableRewards } from "@/view-functions/getClaimableRewards";
+import { getUserHasStake } from "@/view-functions/getUserHasStake";
+import { getUserStakeData } from "@/view-functions/getUserStakeData";
+import { useWallet } from "@aptos-labs/wallet-adapter-react";
+import { useQuery } from "@tanstack/react-query";
+import { PropsWithChildren, createContext, useEffect, useState } from "react";
+
+export interface AccountDataDataProviderState {
+  hasStake: boolean;
+  hasRewards: boolean;
+  claimableRewards: number;
+  accountStakeAmount: number;
+  isCreator: boolean;
+  accountTokenBalance: string;
+}
+
+const defaultValues: AccountDataDataProviderState = {
+  hasStake: false,
+  hasRewards: false,
+  claimableRewards: 0,
+  accountStakeAmount: 0,
+  isCreator: false,
+  accountTokenBalance: "0",
+};
+
+export const AccountDataContext = createContext<AccountDataDataProviderState>(defaultValues);
+
+export const AccountDataContextProvider: React.FC<PropsWithChildren> = ({ children }) => {
+  const { account } = useWallet();
+  const { tokenData } = useGetTokenData();
+  const { toast } = useToast();
+
+  const [hasStake, setHasStake] = useState<boolean>(false);
+  const [hasRewards, setHasRewards] = useState<boolean>(false);
+  const [claimableRewards, setClaimableRewards] = useState<number>(0);
+  const [accountStakeAmount, setAccountStakeAmount] = useState<number>(0);
+  const [isCreator, setIsCreator] = useState<boolean>(false);
+  const [accountTokenBalance, setAccountTokenBalance] = useState<string>("0");
+
+  const { data } = useQuery({
+    queryKey: ["account-data-context", account],
+    refetchInterval: 1000 * 30,
+    queryFn: async () => {
+      try {
+        /**
+         * Get the current connected account claimable rewards
+         */
+        const claimableRewards = await getClaimableRewards(account?.address);
+
+        /**
+         * Determine whether the current connected account has staked
+         */
+        const hasStake = await getUserHasStake(account?.address);
+
+        /**
+         * Get the current connected account stake data
+         */
+        const accountStakeData = await getUserStakeData(account?.address);
```

Comment:
should only call this when `hasStake` is true

Reply:
ok i see inside `getUserStakeData` it returns null if any error happens, but still i think it's good to check to let it only catch unexpected errors